### DISCUSSION TITLE
Colors

### DISCUSSION THREAD

**Discussion by nici22** [ssh-mitm/ssh-mitm#146] [General]
Hi, can you tell me whether it is possible to disable colors in logging part? If yes, can you kindly help me which part should I change on source code on development installation ?

**Comment by manfred-kaiser** [comment]
There is an option for json output. 

To use this option, you must install the latest version from the git repo:

```
pip install -U git+https://github.com/ssh-mitm/ssh-mitm.git
```

Note: if you don't get the latest version from the git repo, uninstall the old version: `pip uninstall ssh-mitm`

To use json output, start the server with following command: `ssh-mitm --log-format json server`


If you want to remove coloring in the text output, you can edit following lines:

https://github.com/ssh-mitm/ssh-mitm/blob/79a49dd1ad28dacfb19c8fce9513a8cb42ab2c9d/sshmitm/cli.py#L159:L179

**Reply by nici22** [reply]
Thanks for the reply
I have another question. I have a problem in sending log files with filebeat. However, filebeat can not read the file. Because ssh-mitm works as a keylogger and logs every keystroke like "tab", "delete", "ctrl+smth". It gives correct output only when we use "cat" command. I mean the output of 'cat' command and file editors (nano, vim, etc.) are different. Do you have any idea to solve this problem?

**Reply by manfred-kaiser** [reply]
The "log-output" is in [scriptreplay format](https://man7.org/linux/man-pages/man1/scriptreplay.1.html). There is an additional file for the user input, because scriptreplay does not show user input for hidden input like passwords.

This format is not intended to store session data in Elasticsearch because it is based on the raw output of the pseudoterminal. A better format to store session data in Elasticsearch would be [asciinema](https://asciinema.org/). This format is based on [JSON](https://github.com/asciinema/asciinema/blob/develop/doc/asciicast-v2.md).

At the moment SSH-MITM does not support asciinema :disappointed:  but there are some scripts to convert scriptreplay to asciinema: https://gist.github.com/izabera/223f58d20be61911e62b

It would be nice to have asciinema support, but I have not much time. Currently I'm working to implement [MOSH](https://mosh.org/) which has a higher priority for me, but if you are interested to implement some features I will try to find some time to help you :wink:

**Reply by nici22** [reply]
Thanks a million for your attention and time. I highly appreciate it.
